pygmt/clib/session.py

```diff
@@ -1237,17 +1240,58 @@ def virtualfile_from_grid(self, grid):
         >>> # The output is: w e s n z0 z1 dx dy n_columns n_rows
 
         """
+        try:
+            assert grid.attrs["geocoord_type"] == 1
+            _gtype = "GMT_GRID_IS_GEO"
+        except (AssertionError, KeyError):
+            _gtype = "GMT_GRID_IS_CARTESIAN"
+
+        try:
+            assert grid.attrs["node_offset"] == 1
+            _registration = "GMT_GRID_PIXEL_REG"
+        except (AssertionError, KeyError):
+            _registration = "GMT_GRID_NODE_REG"
```

Comment:
I've renamed `coord_sys` to `geocoord_type`. Where `1=GMT_GRID_IS_GEO` and `0=GMT_GRID_IS_CARTESIAN`. As for the `node_offset` attribute, `1=GMT_GRID_PIXEL_REG` and `0=GMT_GRID_NODE_REG`.

This is fine for plotting, but a problem we'll face is that these attributes might not get propagated when passing them into non-plotting modules (i.e. data processing ones like `grdcut`). That `pygmt.open()` function is starting to look really appealing now, xref https://github.com/GenericMappingTools/pygmt/pull/492#discussion_r446431644.

Reply:
You know what, I have another idea - make a `gmt` [xarray accessor](http://xarray.pydata.org/en/v0.15.1/internals.html#extending-xarray). I'll open up an issue to describe this further, but for the purposes of this PR, the way we would access the `registration` from the xr.DataArray is via `grid.gmt.registration` instead of through `grid.attrs["registration"]`. The benefit of the [xarray dataset/dataarray accessor](http://xarray.pydata.org/en/v0.15.1/generated/xarray.register_dataset_accessor.html#xarray.register_dataset_accessor) is that it would be more stable than keeping stuff in `attrs`. It also means we won't need a `pygmt.open()` function, and the #494 PR might not be needed anymore.

Reply:
Ok, now that #500 is merged (along with the bump to GMT 6.1.0), this PR should be prime for a good refresh. Will need some time resolve the merge conflicts/clean up the code!